src/student/models/ComparableStudent.java

```diff
@@ -0,0 +1,31 @@
+package student.models;
+
+public class ComparableStudent implements Comparable<ComparableStudent> {
```

Comment:
not sure if I understand why there are two type of students, comparable and not, why just not comparable type in allcases?